icu4c/source/common/dictbe.cpp

```diff
@@ -1342,11 +1353,21 @@ CjkBreakEngine::divideUpDictionaryRange( UText *inText,
         t_boundary.addElement(numCodePts, status);
         numBreaks++;
     } else {
-        for (int32_t i = numCodePts; i > 0; i = prev.elementAti(i)) {
+        t_boundary.addElement(numCodePts, status);
```

Comment:
it seems your change will alternative the current behavior of word break , more than what your project intent to change, is that correct?

Reply:
Actually, it doesn't change.
Before: line-1345 and 1346: for loop and addElement
After: line-1355, 1367: add the first element, for loop, and add the remaining elements when necessary.
When isPhraseBreaking is false, everything is identical.
When isPhraseBreaking is true, skip JP particle and Hiragana.